PBTA Histologies: Integrated molecular subtyping to base histology (7 of N)

kgaonkar6 opened the pull request:
<!--Hi there, thanks for your contribution! Please take a moment to fill out this template to facilitate the review of your pull request.-->

### Purpose/implementation Section

#### What scientific question is your analysis addressing?
Base histology needs to be updated with results from molecular-subtyping-pathology. In this module I'm adding molecular_subtype, integrated_diagnosis, short_histology, broad_histology and Notes.

#### What was your approach?
Read pbta-histologies-base.tsv and add molecular_subtype, integrated_diagnosis, short_histology, broad_histology and Notes. Check for changes, if there are updates we need to re-run summary modules/plots which use this information. Check for duplicates if any, and fix.


#### What GitHub issue does your pull request address?
#858 


### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.

#### Which areas should receive a particularly close look?



#### Is there anything that you want to discuss further?



#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?
Yes


### Results

#### What types of results are included (e.g., table, figure)?
table


#### What is your summary of the results?
Final histology file for v18 release


#### Reproducibility Checklist

<!-- Check all those that apply or remove this section if it is not applicable.-->

- [x] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [x] This analysis has been added to continuous integration.

#### Documentation Checklist

<!-- Please review https://github.com/AlexsLemonade/OpenPBTA-analysis#documenting-your-analysis -->

- [x] This analysis module has a `README` and it is up to date.
- [x] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [x] The analytical code is documented and contains comments.

<!-- IF YOUR PULL REQUEST IS A DATA RELEASE, PLEASE REMOVE THE [HTML COMMENT TAG](https://html.com/tags/comment-tag/) FROM THE SECTION BELOW AND COMPLETE THE CHECKLIST-->

<!-- 
### Data Release Checklist
- [ ] Is the table in doc/data-file-descriptions.md up to date?
- [ ] Is doc/data-format.md up to date?
- [ ] Is doc/release-notes.md up to date?
- [ ] Is download-data.sh up to date?
- [ ] Was download-data.sh tested and did it complete without error?
-->

jharenza commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
````diff
@@ -0,0 +1,106 @@
+
+diff_short_histology
+```
+The above 172 changes occured because of changes in string value assignement
+In compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv sample with broad_histology== `Ependymal tumor` is `EPN`	but in base histology it is `Ependymoma`.
+For samples with broad_histology==`Embryonal tumor`, short_histology is `Embryonal tumor` but in base histlogy it was `ETMR`
````
```suggestion
The above 172 changes occurred because of changes in string value assignment
In `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` sample with broad_histology== `Ependymal tumor` is `EPN`, but in base histology it is `Ependymoma`.
For samples with broad_histology==`Embryonal tumor`, short_histology is `Embryonal tumor` but in base histology it was `ETMR`.
```

jharenza reviewed (commented):
@kgaonkar6 - This looks good and I fixed a few minor typos.

I want to get a weigh-in from @jaclyn-taroni on finalizing the `pbta-histologies.tsv` file. Up until now, we left anything not subtyped as NA for `integrated_diagnosis` because technically, the integrated diagnosis is an integration of the pathology report with molecular data and when clinicians use this data, they will think that the integrated diagnosis considered molecular data. I had some discussions with Cassie Kline (neuro-oncology physician at CHOP) and we had two thoughts/options on this. 
1. For those not subtyped, we can carry over diagnoses information from `pathology_diagnosis` +/- `pathology_free_text_diagnosis` into the `integrated_diagnosis` column so there is complete data for the tumors.
2. We add to the `integrated_diagnosis` column _**only**_ diagnoses derived from molecular data and create another column, perhaps `harmonized_diagnosis`, which contains _either_ an `integrated_diagnosis` or a `pathology_diagnosis`. 

I think I favor 2 because it is more "correct".

This being said, one subtyping modules does not use molecular data: neurocytoma. 

Thoughts?

jaclyn-taroni commented:
I like option 2 

> We add to the `integrated_diagnosis` column _**only**_ diagnoses derived from molecular data and create another column, perhaps `harmonized_diagnosis`, which contains either an `integrated_diagnosis` or a `pathology_diagnosis`.

But when we document things, I think we'd want to phrase it as `harmonized_diagnosis` is the diagnosis in `integrated_diagnosis` or `pathology_diagnosis` when no `integrated_diagnosis` is available (provided I followed correctly, of course).

jharenza commented:
> I like option 2
> 
> > We add to the `integrated_diagnosis` column _**only**_ diagnoses derived from molecular data and create another column, perhaps `harmonized_diagnosis`, which contains either an `integrated_diagnosis` or a `pathology_diagnosis`.
> 
> But when we document things, I think we'd want to phrase it as `harmonized_diagnosis` is the diagnosis in `integrated_diagnosis` or `pathology_diagnosis` when no `integrated_diagnosis` is available (provided I followed correctly, of course).

ok great. and yes, of course! 

cc @kgaonkar6 to update and create this extra column

kgaonkar6 commented:
Sounds good, is there a preference where I update the `integrated_diagnosis` column for Neurocytoma to NA?  Here or in molecular-subtyping-pathology since currently `integrated_diagnosis` is not NA for Neurocytoma in `../molecular-subtyping-pathology/results/compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` which I'm reading in this module to add intergated_diagnosis

kgaonkar6 commented:
I've added the harmonised_diagnosis column which is integrated_diagnosis if present if integrated_diagnosis==NA then it's pathology_diagnosis. pbta-histologies.tsv file is also updated on s3

jashapiro commented:
With the addition of `harmonised_diagnosis`, do we also want to add a column with short values derived from that field, parallel to `short_histologies` that we might use for standard labels for plotting? 

This might also be a good place to address #647.

kgaonkar6 commented:
Just adding this here as well to get describe the process (as I see it for now) we have for histology generation
<img width="611" alt="Screen Shot 2020-12-22 at 12 57 15 PM" src="https://user-images.githubusercontent.com/34580719/102918635-424a3300-4455-11eb-8ae0-fd85e3b4cfd9.png">

cansavvy commented:
You may want to double check how `integrated_diagnosis` is being set up for v18 as well, it has ~700 extra `NA`s now which causes `interaction plots` to break because only one LGAT sample shows up. https://github.com/AlexsLemonade/OpenPBTA-analysis/issues/875#issuecomment-749705328

jharenza commented:
> With the addition of `harmonised_diagnosis`, do we also want to add a column with short values derived from that field, parallel to `short_histologies` that we might use for standard labels for plotting?
> 
> This might also be a good place to address #647.

I think we may need help on this if we want them shorter than `short_histology` (or just update the short histology). With #849 , #647 is being addressed (maybe I left that out of the release notes, in which case, I can add back). Below are all `Other` in `pathology_diagnosis`:


| pathology_free_text_diagnosis                                   | broad_histology                               | short_histology            |
|-----------------------------------------------------------------|-----------------------------------------------|----------------------------|
| atypical lymphoid infiltrate                                    | Benign tumor                                  | Lymphoid infiltrate        |
| dermoid cyst                                                    | Benign tumor                                  | Cyst                       |
| dermoid inclusion cyst                                          | Benign tumor                                  | Cyst                       |
| epidermoid cyst                                                 | Benign tumor                                  | Cyst                       |
| fibroma                                                         | Benign tumor                                  | Fibroma                    |
| hamartoma                                                       | Benign tumor                                  | Hamartoma                  |
| inclusion cyst                                                  | Benign tumor                                  | Cyst                       |
| meningioangiomatosis                                            | Benign tumor                                  | Meningioangiomatosis       |
| ossifying fibroma                                               | Benign tumor                                  | Fibroma                    |
| osteoblastoma                                                   | Benign tumor                                  | Osteoblastoma              |
| perineuroma                                                     | Benign tumor                                  | Perineuroma                |
| prolactinoma                                                    | Benign tumor                                  | Prolactinoma               |
| rathke cleft cyst                                               | Benign tumor                                  | Cyst                       |
| xanthogranuloma                                                 | Benign tumor                                  | Xanthogranuloma            |
| choroid plexus cyst                                             | Choroid plexus tumor                          | Choroid plexus tumor       |
| cortical tubers                                                 | Choroid plexus tumor                          | Choroid plexus tumor       |
| dipg                                                            | Diffuse astrocytic and oligodendroglial tumor | HGAT                       |
| embryonal tumor with multilayer rosettes, ros (who grade iv)    | Embryonal tumor                               | ETMR                       |
| embryonal tumor, nos, congenital type                           | Embryonal tumor                               | Embryonal tumor            |
| ependymoblastoma                                                | Embryonal tumor                               | Embryonal tumor            |
| medullooepithelioma                                             | Embryonal tumor                               | Embryonal tumor            |
| non- langerhans histiocytosis (jxg- cns)                        | Histiocytic tumor                             | JXG                        |
| rosai-dorfman disease                                           | Histiocytic tumor                             | Rosai-dorfman              |
| malignant melanocytic neoplasm                                  | Melanocytic tumor                             | Melanocytoma               |
| fibromyxoid lesion                                              | Mesenchymal non-meningothelial tumor          | Fibromixoid lesion         |
| myofibroblastic tumor                                           | Mesenchymal non-meningothelial tumor          | Myofibroblastic tumor      |
| glioneuronal tumor                                              | Neuronal and mixed neuronal-glial tumor       | Glial-neuronal tumor NOS   |
| neuroepithelial neoplasm                                        | Neuronal and mixed neuronal-glial tumor       | Glial-neuronal tumor NOS   |
| epilepsy, chronic rasmussen encephalitis                        | Non-tumor                                     | Epilepsy                   |
| reactive connective tissue                                      | Non-tumor                                     | Reactive tissue            |
| ganglioneuroma                                                  | Benign tumor                                  | Ganglioneuroma             |
| arteriovenous malformation                                      | Non-tumor                                     | Arteriovenous malformation |
| myxoid spindle cell tumor                                       | Non-CNS tumor                                 | Myxoid spindle cell tumor  |
| embryonal tumor with abundant neuropil and true rosettes (pnet) | Embryonal tumor                               | ETMR                       |

Do these look OK? There will really be lots of small groups, which would mean more colors, but this will outline the samples better than `Other` - or we can choose not to plot certain samples. I am not sure the best way to handle.

jashapiro commented:
> I think we may need help on this if we want them shorter than `short_histology` (or just update the short histology). With #849 , #647 is being addressed (maybe I left that out of the release notes, in which case, I can add back). Below are all `Other` in `pathology_diagnosis`:

Okay, that makes sense. Thanks!

It does seem that the list of short_histology values is more than a bit unwieldy now. We may need to identify some larger groupings to make plotting reasonable. Or at the least standardized abbreviations for the broad_histology column, if there are fewer of those?

jharenza reviewed (commented):
@kgaonkar6 in addition to #878 that we discussed yesterday, I also noticed that the logic for `harmonized_diagnosis` should be modified for the `Other` samples. 

I had the `rules_without_subtype_WIP` tab from[ this google sheet](https://docs.google.com/spreadsheets/d/1fDXt_YODcSAWDvyI5ISBVhUCu4b5-TFCVWMOwiPeMwk/edit#gid=1210552782) noted as "work in progress". In it, I am compiling all of the `Other` diagnoses into a harmonized diagnosis as well as some of the samples which have additional information in the free text (eg: Meningioma) which can match up to WHO diagnoses. Since this is still a WIP, I wonder if we should take care of this in V19 and in the meantime for all `Other` in `pathology_diagnosis`, make the harmonized diagnosis a concatenation of `pathology_diagnosis`, `pathology_free_text_diagnosis` separated by a comma and space? (eg: _Other, malignant melanocytic neoplasm_).

kgaonkar6 commented:
Do we only want to update harmonized_diagnosis or broad_hist and short_hist also for "Other" in pathology_diagnosis?

jharenza commented:
> Do we only want to update harmonized_diagnosis or broad_hist and short_hist also for "Other" in pathology_diagnosis?

Oh yes, all of them. I guess these are not a part of the base file, then? Thanks!

kgaonkar6 commented:
The current base histology used the following files to get broad_histology and short_histology: 
```r
# pathology diagnosis is needed to match tumor samples  to broad/short histology
path_dx <- read_sheet('https://docs.google.com/spreadsheets/d/1fDXt_YODcSAWDvyI5ISBVhUCu4b5-TFCVWMOwiPeMwk/edit#gid=0',range="pathology_diagnosis_for_subtyping") %>%
   dplyr::select(pathology_diagnosis,broad_histology, short_histology) %>%
   write_tsv(file.path(input_dir,"pathology_diagnosis_for_subtyping.tsv"))

# pathology free text diagnosis is needed to match to samples marked as "Other" in pathology_diagnosis
path_free_text <- read_sheet('https://docs.google.com/spreadsheets/d/1fDXt_YODcSAWDvyI5ISBVhUCu4b5-TFCVWMOwiPeMwk/edit#gid=0',range="pathology_free_text_diagnosis_for_subtyping") %>%
   dplyr::select(pathology_free_text_diagnosis,broad_histology, short_histology)%>%
   write_tsv(file.path(input_dir,"pathology_free_text_diagnosis_for_subtyping.tsv"))
```

are you proposing changing the above files to the ghseet in your comment above?

kgaonkar6 commented:
ok I’ve updated subtype-integrate with the new broad/short/harm_dx for pathology_dx== “Other” samples. So now for v18 pbta-histologies.tsv the process is:
<img width="567" alt="Screen Shot 2020-12-24 at 3 21 27 PM" src="https://user-images.githubusercontent.com/34580719/103105428-c63e1f80-45fb-11eb-8548-28bcba0b2dba.png">

cansavvy commented:
Based on the conversation on #869 , I've merged master to here and reverted the fusion_filtering files to what is in master. (We'll have to merge to master again after #868 and #869 are merged to master but the fusion_filtering bit should be fixed).

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
```diff
@@ -0,0 +1,196 @@
+
+### Read molecular-subtyping-pathology results
+
+Reading molecular_subtype,	integrated_diagnosis, short_histology,	broad_histology and	Notes from `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` 
```
This spacing is odd. Is there a reason behind this odd spacing?

cansavvy replied:
If you want to do a quick thing to try to get rid of some of the weird spacing in this notebook, you can run `styler::style_file("analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd")` and see if styler can fix some of these things for you. Beyond that, I don't know that we need to fuss with it. 
...
I just checked, and unfortunately we do not have the `styler` package in the docker container. So don't worry about this, but it could be handy for you elsewhere.

kgaonkar6 replied:
hmm not sure where these spaces came from , checking

kgaonkar6 replied:
the spaces looked fine on localhost, I edited it and looks ok here as well. Let me know if I missed any weird spaces

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
````diff
@@ -0,0 +1,196 @@
+
+```{r}
+
+compiled_subtyping<-read_tsv("../molecular-subtyping-pathology/results/compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv")
````
```suggestion
compiled_subtyping<-read_tsv(file.path("..", "molecular-subtyping-pathology", "results", "compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv")
```
You may want to alter the spacing on my suggestion depending on how it looks.

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
````diff
@@ -0,0 +1,196 @@
+
+```{r}
+
+Other_subtypes <- read_tsv(file.path("input","pathology_dx and pathology_free_text_diagnosis to broad_histology for subtyping module - rules_without_subtype_WIP.tsv"))
````
I'm not a fan of this file name but I'm guessing it comes from somewhere else and we don't have control over that.

kgaonkar6 replied:
umm I could update the name and add the link to the googlesheet tab used, just wanted to keep the name as is in the google sheet for reference.

cansavvy replied:
I think the file name should be changed everywhere or not at all. But if changing it will cause confusion, don't bother, no worries.

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
```diff
@@ -0,0 +1,196 @@
+
+No NAs in broad_histology, short_histology or harmonized_diagnosis
+
+Just a note, integrated_diagnosis == NA is expected for samples where subtyping is not performed or is molecular_subtype == "XYZ,To be classified" because no evidence was provided/available for these samples we are not able to add integrated_diagnosis.
```
```suggestion
Just a note, integrated_diagnosis is expected to be `NA` for samples where subtyping is not performed or if molecular_subtype is "XYZ,To be classified". 
This means no evidence was provided/available for these samples so we are not able to add integrated_diagnosis.
```

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
````diff
@@ -0,0 +1,196 @@
+
+```
+The above 172 changes occurred because of changes in string value assignment
+In `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` sample with broad_histology== `Ependymal tumor` is `EPN`, but in base histology it is `Ependymoma`.
````
```suggestion
In `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` a sample with broad_histology that is `Ependymal tumor` is `EPN`, but in `pbta-histologies-base.tsv` it is `Ependymoma`.
```

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
```diff
@@ -0,0 +1,196 @@
+The purpose of this notebook is to integrate molecular subtyping results from 
+[molecular-subtyping-pathology](https://github.com/AlexsLemonade/OpenPBTA-analysis/tree/master/analyses/molecular-subtyping-pathology) with `pbta-histologies-base.tsv`.
+
+Here we are using v18 `pbta-histologies-base.tsv`
```
Just to clarify, (and sorry if this is written elsewhere, but it may be good to add it here too) is the purpose of `pbta-histologies-base.tsv` that you write the subtyping results there and that in the next version of the release, the results in `pbta-histologies-base.tsv` are added to `pbta-histologies.tsv`?

cansavvy replied:
Aka it seems like `pbta-histologies-base.tsv` is like the working draft?

kgaonkar6 replied:
Right, we can call `pbta-histologies-base.tsv` a working draft. 
In each release there will be a `pbta-histologies-base.tsv` are added to `pbta-histologies.tsv`. The **PBTA Histologies 1-7** PR  use the base histology file which have any integrated_diagnosis,Notes and molecular_subtype as NA through all the molecular-subtyping-XYZ . Through this process we gradually create the `pbta-histologies.tsv` for the same release.

You might have noticed that the modules (other than subtyping) were all using RNA data.
https://github.com/kgaonkar6/OpenPBTA-analysis/blob/760f7a42d253edcaf3341ac0873634e55590e59e/scripts/run-for-subtyping.sh#L13-L32

We only ran these modules because along with the `pbta-histologies-base.tsv` we also needed to update these files with the new RNA samples to use in our molecular subtyping modules.

cansavvy replied:
Can you put some form of your description here in the beginning of this notebook?

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
````diff
@@ -0,0 +1,196 @@
+```
+The above 172 changes occurred because of changes in string value assignment
+In `compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv` sample with broad_histology== `Ependymal tumor` is `EPN`, but in base histology it is `Ependymoma`.
+For samples with broad_histology==`Embryonal tumor`, short_histology is `Embryonal tumor` but in base histology it was `ETMR`.
````
```suggestion
For samples where broad_histology is `Embryonal tumor`, short_histology is also `Embryonal tumor` but in base histology it was `ETMR`.
```

cansavvy reviewed (commented):
I think this all seems reasonable. I still don't understand a lot of the background behind the histology labels but it seems like this PR is doing what it set out to do. Some of the wording things were a tad unclear to me so I just tried to rephrase (if I interpreted the explanations incorrectly, let me know). 

One important question: I think we'd still want this to be added to CI testing?

cansavvy reviewed (commented):
I think this is very close. :+1: I just one description add request (see comment) and my outstanding question about this being added to circle CI testing.

kgaonkar6 commented:
Thanks @cansavvy! I added the module to CI and also added code to read pbta-histologies-base.tsv from data directly when running on testing.

jashapiro commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
```diff
@@ -0,0 +1,229 @@
+
+library(tidyverse)
+
+if (running_in_ci) {
+  data_dir <- "../../data/"
+} else {
+data_dir <- "../../data/release-v18-20201123/"
+}
```
This conditional code should not be needed. The way the system is set up, all files should be linked to `data/` with no subfolder. In CI, this will be the testing files, but in regular use (after running `download-data.sh`) it will be the release version. Unless we want to make this file only work in v18 (I don't think we do) we should not be referring to the subfolder. 
```suggestion
data_dir <- "../../data/"
```

jharenza replied:
hmm - @jashapiro I think we did this because it was being run before CI files were generated. However, also thinking about the process, it is a bit circular. This needs to be run for `pbta-histologies.tsv` to be generated and added to the new release, before the CI files are generated. Thoughts on that?

jashapiro replied:
The biggest concern I have is that somebody might rerun this script when generating v19+ and forget to update this line.  I don't know if this is pervasive through the subtyping files. 

(Of course, with the linking approach it is possible to have the wrong set of files linked, and get a hidden error that way, so neither one is great.)

jharenza replied:
@kgaonkar6 @jashapiro what if we add this as a requirement in the bash script in which the user has to put a release version?

jashapiro replied:
I just looked at some of the other subtyping analyses, and we don't do that for those. I think for consistency with the other analyses, it should simply use `data/` and expect that the correct files have been placed or linked there.

kgaonkar6 replied:
ok sounds good, I'll update the code to always use data

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
```diff
@@ -0,0 +1,210 @@
+- molecular-subtyping-LGAT
+- molecular-subtyping-pathology
+
+we gathere and update molecular-subtype AND integrated_diagnosis AND broad_histology AND short_histology for these histologies.
```
```suggestion
We gathered and updated molecular-subtype AND integrated_diagnosis AND broad_histology AND short_histology for these histologies.
```

cansavvy commented on analyses/molecular-subtyping-integrate/01-integrate-subtyping.Rmd:
```diff
@@ -0,0 +1,210 @@
+
+we gathere and update molecular-subtype AND integrated_diagnosis AND broad_histology AND short_histology for these histologies.
+
+In this notebook we will add the molecular subtyping done as part as OpenPBTA to the base histology file to create the `pbta-histologies.tsv` for the same release. 
```
```suggestion
In this notebook, we will add the molecular subtyping updated in the `pbta-histologies-base.tsv` file to update the `pbta-histologies.tsv` file. 
```

kgaonkar6 replied:
> subtyping updated in the `pbta-histologies-base.tsv` 

This is not entirely true I mean `pbta-histologies-base.tsv` file is used but is not updated through the subtyping modules. The subtyping file is saved in 
compiled_subtyping<-read_tsv(file.path("..", "molecular-subtyping-pathology", "results", "compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv")) as part of the molecular-subtyping-pathology module.

We directly use compiled_molecular_subtypes_with_clinical_pathology_feedback.tsv to update `pbta-histologies.tsv` file

cansavvy replied:
Ah. Okay. Can you describe that here then?

kgaonkar6 replied:
oh I forgot to commit the description 😅

Let me know if it makes sense, should I add in my workflow diagram if it's useful in the README or code or this looks ok ?

cansavvy replied:
Sure! Why not add the diagram! Would probably be good to have that in this description.

kgaonkar6 replied:
done!

cansavvy reviewed (approved):
I think this looks fine to me now. Just noticed some typos so if we get those fixed I think this can be merged.
